From 5b488cc802c23c76d100422df249cdbf272bf65f Mon Sep 17 00:00:00 2001
From: Scott Tsai <scottt.tw@gmail.com>
Date: Thu, 10 Apr 2025 18:13:02 +0800
Subject: [PATCH 1/2] Disable conv_wino_fury_RxS for gfx115x

MIOpenDriver convfp16 would produce the folloing error on the gfx1151:

```
kernel: [drm:gfx_v11_0_bad_op_irq [amdgpu]] *ERROR* Illegal opcode in command stream
stderr: HSA_STATUS_ERROR_INVALID_ISA: The instruction set architecture is invalid. code: 0x100f
```
---
 src/solver/conv/conv_wino_fury_RxS.cpp | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/solver/conv/conv_wino_fury_RxS.cpp b/src/solver/conv/conv_wino_fury_RxS.cpp
index 6f90642fa..66cb1b4d8 100644
--- a/src/solver/conv/conv_wino_fury_RxS.cpp
+++ b/src/solver/conv/conv_wino_fury_RxS.cpp
@@ -255,6 +255,13 @@ bool ConvWinoFuryRxSCommon<Winodata, Winofilter>::IsApplicable(const ExecutionCo
     if(!StartsWith(dev_name, "gfx11"))
         return false;
 
+    if(StartsWith(dev_name, "gfx115")) {
+        // Triggers this error on gfx1151
+        // kernel: [drm:gfx_v11_0_bad_op_irq [amdgpu]] *ERROR* Illegal opcode in command stream
+        // stderr: HSA_STATUS_ERROR_INVALID_ISA: The instruction set architecture is invalid. code: 0x100f
+        return false;
+    }
+
     if(!(problem.GetKernelStrideH() == 1 && problem.GetKernelStrideW() == 1))
         return false;
     if(!(problem.GetDilationH() == 1 && problem.GetDilationW() == 1))
-- 
2.43.0

